dan_layer/epoch_manager/src/base_layer/base_layer_epoch_manager.rs

```diff
@@ -461,16 +461,24 @@ impl BaseLayerEpochManager<SqliteGlobalDbAdapter, GrpcBaseNodeClient> {
         &self,
         epoch: Epoch,
     ) -> Result<ValidatorNodeBalancedMerkleTree, EpochManagerError> {
+        let db = {
+            let mut tx = self.global_db.create_transaction()?;
+            self.global_db.bmt(&mut tx).get_bmt(epoch.0)
+        }?;
+        if let Some(bmt) = db {
+            return Ok(bmt);
+        }
         let vns = self.get_validator_nodes_per_epoch(epoch)?;
 
-        // TODO: the MMR struct should be serializable to store it only once and avoid recalculating it every time per
-        // epoch
         let mut vn_bmt_vec = Vec::with_capacity(vns.len());
         for vn in vns {
             vn_bmt_vec.push(vn.node_hash().to_vec())
         }
 
-        let vn_bmt = ValidatorNodeBalancedMerkleTree::create(vn_bmt_vec);
+        let vn_bmt = ValidatorNodeBalancedMerkleTree::create(vn_bmt_vec.clone());
```

Comment:
Dont think this clone is necessary
```suggestion
        let vn_bmt = ValidatorNodeBalancedMerkleTree::create(vn_bmt_vec);
```